internal/pkg/api/event.go

```diff
@@ -0,0 +1,103 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package api
+
+import (
+	"fmt"
+	"time"
+)
+
+// The event type.
+const (
+	// Mapping to the type supported by SkyWalking OAP.
+	_ EventType = iota
+	MetricsEvent
+	ProflingEvent
+	SegmentEvent
+	ManagementEvent
+	MeterEvent
+	LogEvent
+)
+
+type EventType int32
+
+// Event that implement this interface would be allowed to transmit in the Satellite.
+type Event interface {
+	// Name is a identify to distinguish different events.
+	Name() string
```

Comment:
If it's an identifier, maybe name it as `ID` is more semantic?

Reply:
Sorry for the comment, `type` is a sign to distinguish different events